packages/app-builder/src/routes/ressources+/analytics+/$scenarioId+/query.ts

```diff
@@ -1,60 +1,47 @@
-import { initServerServices } from '@app-builder/services/init.server';
-import { getRoute } from '@app-builder/utils/routes';
-import { type ActionFunctionArgs } from '@remix-run/node';
+import { createServerFn } from '@app-builder/core/requests';
+import { authMiddleware } from '@app-builder/middlewares/auth-middleware';
+import { handleRedirectMiddleware } from '@app-builder/middlewares/handle-redirect-middleware';
+import { analyticsQuery } from '@app-builder/models/analytics';
 import { z } from 'zod';
 
 const urlParamsSchema = z.object({
   scenarioId: z.uuidv4(),
 });
 
-const queryParamsSchema = z.object({
-  dateRange: z.object({
-    start: z.iso.datetime(),
-    end: z.iso.datetime(),
-  }),
-  compareDateRange: z
-    .object({
-      start: z.iso.datetime(),
-      end: z.iso.datetime(),
-    })
-    .optional(),
-  scenarioVersion: z.number().optional(),
-  trigger: z
-    .array(
-      z.object({
-        field: z.uuidv4(),
-        op: z.enum(['=', '!=', '>', '>=', '<', '<=']),
-        values: z.array(z.string()),
-      }),
-    )
-    .optional()
-    .default([]),
-});
-
-export async function action({ params, request }: ActionFunctionArgs) {
-  const { authService } = initServerServices(request);
-  const { analytics } = await authService.isAuthenticated(request, {
-    failureRedirect: getRoute('/sign-in'),
-  });
+export const action = createServerFn(
+  [handleRedirectMiddleware, authMiddleware],
+  async function createAnalyticsQueryAction({ params, request, context }) {
+    try {
+      const urlParams = urlParamsSchema.parse(params);
 
-  const urlParams = urlParamsSchema.parse(params);
+      const body = await request.json();
 
-  const body = await request.json();
-  const queryParams = queryParamsSchema.parse(body);
+      const queryParams = analyticsQuery.parse({
+        ...body,
+        scenarioId: urlParams.scenarioId,
+      });
 
-  const [decisionOutcomesPerDay, ruleHitTable] = await Promise.all([
-    await analytics.getDecisionOutcomesPerDay({
-      ...queryParams,
-      scenarioId: urlParams.scenarioId,
-    }),
-    await analytics.getRuleHitTable({
-      ...queryParams,
-      scenarioId: urlParams.scenarioId,
-    }),
-  ]);
+      const [decisionOutcomesPerDay, ruleHitTable] = await Promise.all([
+        await context.authInfo.analytics.getDecisionOutcomesPerDay({
+          ...queryParams,
+          scenarioId: urlParams.scenarioId,
+        }),
+        await context.authInfo.analytics.getRuleHitTable({
+          ...queryParams,
+          scenarioId: urlParams.scenarioId,
+        }),
```

Comment:
No `await` in there cause `Promise.all` should be an array of promises

Reply:
you are right, we are awaiting the Promise.all already